components/views.py

```diff
@@ -57,10 +76,28 @@ def list(self, request, *args, **kwargs):
 
 class ComponentTypeListView(generics.ListAPIView):
     queryset = Component.objects.order_by().values_list("type").distinct()
-    permission_classes = [ComponentPermissions, ]
-    filter_backends = [ComponentPermissionsFilter, ]
+    permission_classes = [
+        ComponentPermissions,
+    ]
+    filter_backends = [
+        ComponentPermissionsFilter,
+    ]
 
     def list(self, request, *args, **kwargs):
         queryset = self.filter_queryset(self.get_queryset())
 
         return Response(queryset, status=status.HTTP_200_OK)
+
+
+class ComponentNarrativeView(generics.RetrieveUpdateAPIView):
+    queryset = Component.objects.all()
+    permission_classes = [
+        ComponentPrivatePermissions,
+    ]
+    serializer_class = ComponentControlSerializer
+
+    # def patch(self, request, *args, **kwargs):
+    #     return self.partial_update(request, *args, **kwargs)
+
+    def patch(self, request, *args, **kwargs):
+        return self.update(request, *args, **kwargs)
```

Comment:
I don't think it's a good idea mixing and matching like this. Everywhere else where we use `generics`, a patch will allow partial updates, but this won't.